Sources/ForageSDK/Component/ForagePINTextField/Protocol/PINVaultTextField.swift

```diff
@@ -0,0 +1,339 @@
+        }
+    }
+    
+    var padding: UIEdgeInsets {
+        get {
+            return textField.padding
+        }
+        set {
+            textField.padding = newValue
+        }
+    }
+    
+    var borderColor: UIColor? {
+        get {
+            return textField.borderColor
+        }
+        set {
+            textField.borderColor = newValue
+        }
+    }
+    
+    
+    func setAccessibilityIdentifier(_ identifier: String) {
+        textField.accessibilityIdentifier = identifier
+    }
+    
+    func setIsAccessibilityElement(_ flag: Bool) {
+        textField.isAccessibilityElement = flag
+    }
+    
+    func setPlaceholder(_ text: String) {
+        textField.placeholder = text
+    }
+
+    
+    
+    var textColor: UIColor? {
+        get { return textField.textColor }
+        set { textField.textColor = newValue }
+    }
+    
+    var font: UIFont? {
+        get { return textField.font }
+        set { textField.font = newValue }
+    }
+    
+    var isSecureTextEntry: Bool {
+        get { return textField.isSecureTextEntry }
+        set { textField.isSecureTextEntry = newValue }
+    }
+    
+    var textAlignment: NSTextAlignment {
+        get { return textField.textAlignment }
+        set { textField.textAlignment = newValue }
+    }
+    
+}
+
+class BasisTheoryTextFieldWrapper: UIView, UITextFieldDelegate, PINVaultTextField{
+    var collector: VaultCollector
+    
+    weak var delegate: PINVaultTextFieldDelegate?
+    
+    func isValid() -> Bool {
+        return textField.metadata.valid
+    }
+    
+    @objc func btTextFieldDidChange(_ textField: UITextField) {
+            delegate?.textFieldDidChange(self)
+    }
+    
+    var placeholder: String?
+    
+    var tfTintColor: UIColor?
+    
+    var verticalConstraint: [NSLayoutConstraint]?
+    var horizontalConstraints: [NSLayoutConstraint]?
+    
+    private let textField: TextElementUITextField
+    
+    
+    override init(frame: CGRect) {
+        textField = TextElementUITextField()
+        collector = CollectorFactory.createBasisTheory(environment: ForageSDK.shared.environment, textElement: textField)
+        
+        super.init(frame: frame)
+        
+        commonInit()
+    }
+    
+    required init?(coder aDecoder: NSCoder) {
+        textField = TextElementUITextField()
+        collector = CollectorFactory.createBasisTheory(environment: ForageSDK.shared.environment, textElement: textField)
+        super.init(coder: aDecoder)
+        
+        commonInit()
+    }
+    
+    private func commonInit() {
+        addSubview(textField)
+        textField.translatesAutoresizingMaskIntoConstraints = false
+        
+        let regexDigit = try! NSRegularExpression(pattern: "\\d")
+        
+        let pinMask =   [
+            regexDigit,
+            regexDigit,
+            regexDigit,
+            regexDigit
+        ] as [Any]
+        
+        let phoneNumberRegex = try! NSRegularExpression(pattern: "^\\d{4}$")
```

Comment:
Out of curiosity, why is this called `phoneNumberRegex`?

Reply:
cuz i copy-pasta'd too hard from bt docs